File: 2-Open-Closed-Principle (OCP)/python/OCP-good-example.py
# Python Good Example
class Post:
    def __init__(self, content):
        self.content = content

    def publish(self):
        # Logic to publish the post
        pass

class ImagePost(Post):
    def __init__(self, content, image_url):
        super().__init__(content)
        self.image_url = image_url

    def publish(self):
        # Logic specific to publishing an image post
        pass

File: 2-Open-Closed-Principle (OCP)/js/OCP-good-example.js
// JavaScript Good Example
class Post {
    constructor(content) {
      this.content = content;
    }
  
    publish() {
      // Logic to publish the post
    }
  }
  
  class ImagePost extends Post {
    constructor(content, imageUrl) {
      super(content);
      this.imageUrl = imageUrl;
    }
  
    publish() {
      // Logic specific to publishing an image post
    }
  }

  class TextPost extends Post {
      constructor(content, description) {
        super(content);
        this.drescription = description;
      }
    
      publish() {
        // Logic specific to publishing an image post
      }
  }